analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.cs

```diff
@@ -115,4 +115,14 @@ public void Foo(IEnumerable enumerable, IEnumerable<object> genericEnumerable)
 
         public void OfType() => OfType();
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/9498
+    public class Repro_9498
+    {
+        void Sample()
+        {
+            float x = 3.14f;
+            float res = 1 / (float)(x * 2); // Noncompliant FP
```

Comment:
Can you find an example where the precision change is visible?

Reply:
I don't think it matters, as the solution would probably be what Jared suggested on the post.
Which is to be conservative and raise only in very few circumstances, when the RHS is a variable.

Even more conservative, we could also not raise on floating point arithmetic at all. 

What do you think?

Reply:
I think we should document a case where this is actually an FP and think about a concrete solution later (or consider it a niche case and ignore it).